From 2b4b4e83e24340768276a1beeb40bd6808db122b Mon Sep 17 00:00:00 2001
From: Sergio Schvezov <sergio.schvezov@canonical.com>
Date: Thu, 12 Dec 2013 00:29:45 -0300
Subject: [PATCH] Installing go when DH_GOLANG_INSTALL_ALL not set

Changed to use a more positive logic and made the DH_GOLANG_INSTALL_ALL
export behave similar to DH_VERBOSE.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>
---
 lib/Debian/Debhelper/Buildsystem/golang.pm | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/Debian/Debhelper/Buildsystem/golang.pm b/lib/Debian/Debhelper/Buildsystem/golang.pm
index 9d55e63..82251cf 100644
--- a/lib/Debian/Debhelper/Buildsystem/golang.pm
+++ b/lib/Debian/Debhelper/Buildsystem/golang.pm
@@ -65,9 +65,9 @@ sub configure {
         },
         wanted => sub {
             my $name = $File::Find::name;
-            return if substr($name, -3) eq '.go' &&
-                      (!exists($ENV{DH_GOLANG_INSTALL_ALL}) ||
-                       $ENV{DH_GOLANG_INSTALL_ALL} != 1);
+            return unless (substr($name, -3) eq '.go' ||
+                      (exists($ENV{DH_GOLANG_INSTALL_ALL}) &&
+                       $ENV{DH_GOLANG_INSTALL_ALL} ne ""));
             return unless -f $name;
             # Store regexp/utf.go instead of ./regexp/utf.go
             push @sourcefiles, substr($name, 2);
-- 
1.8.5.1

